### PR TITLE
feat: Add search to contact messages list

### PR DESCRIPTION
**Pull request by swapnil-turing**
- Search contact messages by name using ?search= query param
- Add tests for search match and no-match cases

# Fixes Issue

**My PR closes #issue_number_here**

# 👨‍💻 Changes proposed(What did you do ?)

# ✔️ Check List (Check all the applicable boxes)

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/contact/tests.py [hunk 1]
```diff
@@ -144,3 +144,31 @@ def test_create_contact_message_ignores_status(self):
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         contact_message = ContactMessage.objects.get(name="Test Name")
         self.assertEqual(contact_message.status, ContactMessage.Status.UNREAD)
+
+    def test_search_contact_messages_by_name(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/tests.py]: value:good-but-wont-fix; category:documentation; feedback:A valid housekeeping point to update PR placeholders and fix checklist rendering prior to merge.

File: augment-store/server/contact/views.py [hunk 2]
```diff
@@ -38,6 +38,15 @@ class ContactListView(CachedListMixin, BaseContactView, ListAPIView):
     cache_service_class = ContactCacheService
     cache_ttl = 60 * 5  # 5 minutes - short TTL due to PII content
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        search = self.request.query_params.get('search')
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/views.py:45]: value:useful; category:bug; feedback:Stripping whitespace prevents edge cases where empty or whitespace-only searches return all results unexpectedly.

File: augment-store/server/contact/tests.py [hunk 3]
```diff
@@ -144,3 +144,31 @@ def test_create_contact_message_ignores_status(self):
...
+            message="I have a question",
+        )
+        url = reverse("v1:contact_list")
+        response = self.authenticated_client.get(url, {"search": "John"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        results = response.data.get("results", [])
+        self.assertGreater(len(results), 0)
+        self.assertIn("John", results[0]["name"])
+
+    def test_search_contact_messages_no_match(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/tests.py:196]: value:useful; category:bug; feedback:Pointed out that the no-match test was passing vacuously; adding a fixture ensures the filter is genuinely preventing non-matching records from being returned.

File: augment-store/server/contact/tests.py [hunk 4]
```diff
@@ -144,3 +144,31 @@ def test_create_contact_message_ignores_status(self):
...
+            name="Jane Smith",
+            email="jane@example.com",
+            subject="Question",
+            message="I have a question",
+        )
+        url = reverse("v1:contact_list")
+        response = self.authenticated_client.get(url, {"search": "John"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        results = response.data.get("results", [])
+        self.assertGreater(len(results), 0)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/tests.py]: value:useful; category:bug; feedback:Added an explicit check for the number of results to ensure no unexpected matches slip through, making the test strictly verify the filter's correctness.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/contact/tests.py [hunk 5]
```diff
@@ -144,3 +144,56 @@ def test_create_contact_message_ignores_status(self):
...
+        ContactMessageFactory(
+            name="Bob Ross",
+            email="bob@example.com",
+            subject="Painting",
+            message="Happy little trees",
+        )
+        url = reverse("v1:contact_list")
+        
+        # Test search by name (John)
+        response = self.authenticated_client.get(url, {"search": "John"})
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/tests.py]: value:useful; category:bug; feedback:Clearing the cache namespace in test setup prevents test leakage and flakiness when using LocMemCache, ensuring reliable CI runs.

File: augment-store/server/contact/views.py [hunk 6]
```diff
@@ -38,6 +38,20 @@ class ContactListView(CachedListMixin, BaseContactView, ListAPIView):
     cache_service_class = ContactCacheService
     cache_ttl = 60 * 5  # 5 minutes - short TTL due to PII content
 
+    def get_queryset(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/views.py:41]: value:delightful; category:bug; feedback:Bumping the cache version is a critical step when changing query parameter semantics on a cached endpoint, preventing stale or incorrectly filtered data from being served after deployment.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/contact/tests.py [hunk 7]
```diff
@@ -144,3 +146,56 @@ def test_create_contact_message_ignores_status(self):
...
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/contact/tests.py]: value:useful; category:bug; feedback:Ensured that testing a name-only search actually exercises the name field independently of overlapping email domain names, preventing a false positive pass.

**Review by michaelos443 — APPROVED** [review]
